diff --git a/dbgen/dss.ddl b/dbgen/dss.ddl
index 89efe56..d765a86 100644
--- a/dbgen/dss.ddl
+++ b/dbgen/dss.ddl
@@ -51,16 +51,18 @@ CREATE TABLE ORDERS  ( O_ORDERKEY       INTEGER NOT NULL,
                            O_SHIPPRIORITY   INTEGER NOT NULL,
                            O_COMMENT        VARCHAR(79) NOT NULL);
 
+CREATE TYPE line_enum AS ENUM ('F', 'O');
+CREATE TYPE ret_enum AS ENUM ('N', 'A', 'R');
 CREATE TABLE LINEITEM ( L_ORDERKEY    INTEGER NOT NULL,
                              L_PARTKEY     INTEGER NOT NULL,
                              L_SUPPKEY     INTEGER NOT NULL,
                              L_LINENUMBER  INTEGER NOT NULL,
-                             L_QUANTITY    DECIMAL(15,2) NOT NULL,
-                             L_EXTENDEDPRICE  DECIMAL(15,2) NOT NULL,
-                             L_DISCOUNT    DECIMAL(15,2) NOT NULL,
-                             L_TAX         DECIMAL(15,2) NOT NULL,
-                             L_RETURNFLAG  CHAR(1) NOT NULL,
-                             L_LINESTATUS  CHAR(1) NOT NULL,
+                             L_QUANTITY    double precision NOT NULL,
+                             L_EXTENDEDPRICE  double precision NOT NULL,
+                             L_DISCOUNT    double precision NOT NULL,
+                             L_TAX         double precision NOT NULL,
+                             L_RETURNFLAG  ret_enum NOT NULL,
+                             L_LINESTATUS  line_enum NOT NULL,
                              L_SHIPDATE    DATE NOT NULL,
                              L_COMMITDATE  DATE NOT NULL,
                              L_RECEIPTDATE DATE NOT NULL,
